bash.sh

```diff
@@ -19,20 +19,19 @@ if [ -d $venv ]; then
     source "$git_base_dir/.venv/bin/activate"
     
     # if the requirements have not changed since the last install
-    if cmp -s requirements.txt $git_base_dir/.venv/bin/requirements.txt; then
+    if cmp -s pyproject.toml $git_base_dir/.venv/bin/pyproject.toml; then
         echo "venv is up to date"
-    else # i.e. the requiremnts have changed
-       pip install -r requirements.txt
+    else # i.e. the requiremnts may have changed
+       poetry install
```

Comment:
again, let's make sure we are using all of these executables with their absolute paths

Reply:
Wasn't the main point of this PR so I didn't throw that in, but definitely an issue to do in the future.